sinks_sentry.go

```diff
@@ -24,48 +30,51 @@ type sentrySink struct {
 	core *sentrycore.Core
 }
 
-// NewSentrySink instantiates a Sentry sink to provide to `log.Init` with default options.
-//
-// See sentrycore.DefaultSentryClientOptions for the default options.
+// NewSentrySink instantiates a Sentry sink to provide to `log.Init` with the following default values:
+// - SampleRate: 0.1
+// To provide different values see `NewSentrySinkWith`
 func NewSentrySink() Sink {
-	return &sentrySink{SentrySink: SentrySink{options: sentrycore.DefaultSentryClientOptions}}
+	return &sentrySink{SentrySink: SentrySink{SampleRate: 0.1}}
 }
 
-// NewSentrySinkWithOptions instantiates a Sentry sink with advanced initial configuration
-// to provide to `log.Init`. Note that configuration, notably the Sentry DSN, may be
-// overwritten by subsequent calls to the `Update` callback from `log.Init`.
-//
-// See sentrycore.DefaultSentryClientOptions for the default options.
-func NewSentrySinkWithOptions(opts sentry.ClientOptions) Sink {
-	return &sentrySink{SentrySink: SentrySink{options: sentrycore.ApplySentryClientDefaultOptions(opts)}}
+// NewSentrySinkWith instantiates a Sentry sink to provide to `log.Init` with the values provided in SentrySink
+func NewSentrySinkWith(s SentrySink) Sink {
+	return &sentrySink{SentrySink: s}
 }
 
 func (s *sentrySink) Name() string { return "SentrySink" }
 
 func (s *sentrySink) build() (zapcore.Core, error) {
-	opts := s.SentrySink.options
-	opts.Dsn = s.DSN
-	client, err := sentry.NewClient(opts)
+	client, err := sentry.NewClient(s.clientOptions())
 	if err != nil {
 		return nil, err
 	}
 	s.core = sentrycore.NewCore(sentry.NewHub(client, sentry.NewScope()))
 	return s.core, nil
 }
 
+func (s *sentrySink) clientOptions() sentry.ClientOptions {
+	return sentry.ClientOptions{
+		Dsn:        s.DSN,
+		SampleRate: s.SampleRate,
+		Debug:      s.Debug,
+	}
+}
+
 func (s *sentrySink) update(updated SinksConfig) error {
-	var updatedDSN string
-	if updated.Sentry != nil {
-		updatedDSN = updated.Sentry.DSN
+	if updated.Sentry == nil {
+		return nil
 	}
 
-	if s.DSN == updatedDSN {
+	if s.DSN == updated.Sentry.DSN && s.SampleRate == updated.Sentry.SampleRate && s.Debug == updated.Sentry.Debug {
```

Comment:
nit: can we find a better way to handle this? This is going to make us think it's working whereas it's not if we use a sentry option that isn't included here

Reply:
Catching up here a bit. Why do we specifically export a few options and not just all the options ? Why can't we use something like:
```
type SentrySink struct {
   sentry.ClientOptions
}
```

and for the above comparison
```
if cmp.Equals(s.ClientOptions, updated.Sentry.ClientOptions) {
   return nil
}
```

Reply:
You're right, that'd be much simpler!